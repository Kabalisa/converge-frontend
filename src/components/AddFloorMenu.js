import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'react-apollo';
import { IconMenu, MenuItem, MenuDivider } from 'react-toolbox/lib/menu';
import AddFloorComponent from './AddFloor';
import { GET_ALL_OFFICES_QUERY } from '../graphql/queries/Offices';
import Spinner from './commons/Spinner';

/**
 * Renders the add office button
 *
 * @returns {JSX}
 */
const menuText = () => (
  <div className="addRoomBtn">
    <span>Add Floor</span>
  </div>
);

/**
 * Drop down list of offices to which floors can be added
 *
 * @returns {JSX}
 */
export class AddFloorMenu extends React.Component {
  static propTypes = {
    data: PropTypes.shape({
      allOffices: PropTypes.shape({
        office: PropTypes.array,
      }),
      loading: PropTypes.bool,
      error: PropTypes.object,
    }),
    refetch: PropTypes.func.isRequired,
  };

  static defaultProps = {
    data: {},
  };

  /**
 * Loops through all offices and returns each one of them as a menu item.
 *
 * @returns {JSX}
 */
  getOffice = () => {
    const { loading, allOffices: { offices } = {}, error } = this.props.data;
    if (loading) return <Spinner />;
    if (error) return <div>{error.message}</div>;
    const officeList = offices.filter(office => office.blocks.length !== 0);
    return (
      officeList.map(office => (
        <MenuItem key={office.id}>
          <AddFloorComponent
            theOffice={office.name}
            refetch={this.props.refetch}
            blocks={office.blocks}
          />
        </MenuItem>
      ))
    );
  }

  render() {
    return (
      <IconMenu className="add-room-menu" icon={menuText()}>
        <MenuItem caption="Select Office" disabled />
        <MenuDivider />
        {this.getOffice()}
      </IconMenu>
    );
  }
}

export default graphql(GET_ALL_OFFICES_QUERY, {
  options: () => ({
    variables: {
      page: 1,
      perPage: 50,
    },
  }),
})(AddFloorMenu);
