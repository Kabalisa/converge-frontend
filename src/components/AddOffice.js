import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import toastr from 'toastr';
import MrmModal from '../components/commons/Modal';
import { Input, SelectInput as Select } from './commons';

import '../assets/styles/addoffice.scss';
import ActionButtons from './commons/ActionButtons';
import { GET_LOCATIONS_QUERY } from '../graphql/queries/Rooms';
import ADD_OFFICE_MUTATION from '../graphql/mutations/offices';
import notification from '../utils/notification';

/**
 * Add Office Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
export class AddOffice extends Component {
  static propTypes = {
    locations: PropTypes.shape({
      allLocations: PropTypes.array,
    }),
    addOffice: PropTypes.func.isRequired,
  };

  static defaultProps = {
    locations: {},
  };
  state = {
    officeName: '',
    officeLocation: '',
    closeModal: false,
    isLoading: false,
  };

  /**
   * Ensures that the modal for adding office closes
   * when a user hits CANCEL on the modal or when the
   * creation of an office is successful
   *
   * @returns {void}
   */

  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * Ensures that the state is updated basing on the
   * user input
   *
   * @param {object} event The input event parameter
   *
   * @returns {void}
   */
  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  /**
   * It updates the state value of closeModal
   * to false whenever the modal closes
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  /**
   * 1. Validates the user input
   * 2. Submits the office data to the backend
   * 3. Notifies the user about the response from his request
   * i.e whether it was a success or a failure
   *
   * @param {object} event
   *
   * @returns {Function}
   */
  handleAddOffice = (event) => {
    event.preventDefault();
    const { officeLocation, officeName } = this.state;
    if (!officeName) {
      notification(toastr, 'error', 'Office name is required')();
    } else if (!officeLocation) {
      notification(toastr, 'error', 'Office location is required')();
    } else {
      this.toggleLoading();
      this.props
        .addOffice({
          variables: {
            locationId: officeLocation,
            name: officeName,
          },
        })
        .then((office) => {
          this.toggleLoading();
          this.handleCloseModal();
          const { name } = office.data.createOffice.office;
          notification(
            toastr,
            'success',
            `${name} office has been added successfully`,
          )();
        })
        .catch((err) => {
          this.toggleLoading();
          this.handleCloseModal();
          notification(toastr, 'error', err.graphQLErrors[0].message)();
        });
    }
  };

  /**
   * 1. change isLoading state to it's opposite value
   * i.e true to false or vise verser
   *
   * @returns {void}
   */
  toggleLoading = () => {
    this.setState({
      isLoading: !this.state.isLoading,
    });
  }

  render() {
    const {
      officeName, officeLocation, closeModal, isLoading,
    } = this.state;
    const { allLocations } = this.props.locations;

    return (
      <MrmModal
        title="ADD OFFICE"
        buttonText="Add Office"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="add-office-modal"
        modalButton="add-button"
      >
        <form className="modal-form">
          <Input
            labelName="Office Name"
            name="officeName"
            value={officeName}
            placeholder="Enter office name"
            id="officeName"
            onChange={this.handleInputChange}
          />
          <Select
            labelText="Select Office Location"
            name="officeLocation"
            id="officeLocation"
            value={officeLocation}
            onChange={this.handleInputChange}
            wrapperClassName="input-wrapper"
            placeholder="Select office location"
            options={allLocations}
          />
          <ActionButtons
            withCancel
            onClickCancel={this.handleCloseModal}
            isLoading={isLoading}
            actionButtonText="ADD OFFICE"
            onClickSubmit={this.handleAddOffice}
          />
        </form>
      </MrmModal>
    );
  }
}

export default compose(
  graphql(GET_LOCATIONS_QUERY, { name: 'locations' }),
  graphql(ADD_OFFICE_MUTATION, { name: 'addOffice' }),
)(AddOffice);
