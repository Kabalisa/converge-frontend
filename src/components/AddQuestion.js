/* eslint react/no-array-index-key: 0 */
import React, { Component } from 'react';
import { Button } from 'react-toolbox/lib/button';
import moment from 'moment';
import Calendar from './commons/Calendar';
import MrmModal from '../components/commons/Modal';
import { SelectInput as Select } from './commons';
import '../assets/styles/addQuestion.scss';

/**
 * Add Question Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
class AddQuestion extends Component {
  state = {
    questionType: '',
    startDate: moment().format('MMM DD Y'),
    endDate: moment().format('MMM DD Y'),
    startTime: moment().format('HH:MM'),
    endTime: moment()
      .add(1, 'hours')
      .format('HH:MM'),
    calenderOpen: false,
    closeModal: false,
    options: [
      {
        id: '1',
        name: 'Rate',
      },
      {
        id: '2',
        name: 'Check',
      },
      {
        id: '3',
        name: 'Suggest',
      },
    ],
  };

  /**
   * It toggles the visibility of the modal
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * It updates the state value of closeModal to false
   * whenever the modal closes
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  /**
   * It toggles the visibility of the calendar modal
   *
   * @returns {void}
   */
  calenderToggle = () => {
    const { calenderOpen } = this.state;
    this.setState({ calenderOpen: !calenderOpen });
  };

  /**
   * It updates the state with the selected start and end date
   *
   * @param {string} start
   * @param {string} end
   *
   * @returns {void}
   */
  sendDateData = (start, end) => {
    this.setState({ startDate: start, endDate: end });
    this.calenderToggle();
  };

  /**
   * It shows the start and end date in the calendar button
   *
   * @returns {JSX}
   */
  calendarIcon = () => (
    <div className="calendarIconBtn">
      <span>{`${this.state.startDate}  -  ${this.state.endDate}`}</span>
    </div>
  );

  /**
   * It handles the selected choice from the options
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  /**
   * It returns the action buttons for the question modal
   *
   * @returns {JSX}
   */
  renderActionButtons = () => (
    <div className="button-container">
      <button onClick={this.handleCloseModal}>save question</button>
      <button onClick={this.handleCloseModal}>cancel</button>
    </div>
  );

  /**
   * It returns the inputs for adding start time and endtime of a question
   * @param {Function} handleInputChange
   *
   * @returns {JSX}
   */
  renderTimeInputs = () => (
    <div className="inputs-inline">
      <div className="inputs">
        <label htmlFor="startTime" className="inputs-title">
          <span>Start</span>
        </label>
        <input
          type="time"
          name="startTime"
          onChange={this.handleInputChange}
          id="timeField"
          value={this.state.startTime}
        />
      </div>
      <div className="inputs">
        <label htmlFor="endTime" className="inputs-title">
          <span>End</span>
        </label>
        <input
          type="time"
          name="endTime"
          onChange={this.handleInputChange}
          id="timeField"
          value={this.state.endTime}
        />
      </div>
    </div>
  );

  /**
   * It returns the input text box for typing the question
   *
   * @returns {JSX}
   */
  renderQuestionInputBox = () => (
    <div className="inputs">
      <input
        placeholder="Enter Question here..."
        type="text"
        name="question"
        onChange={this.handleInputChange}
      />
    </div>
  );

  /**
   * It shows all the input fields required for creating a question
   *
   * @returns {JSX}
   */
  renderQuestionValues = () => (
    <div>
      <span className="question-form__sections">Date</span>
      <div>
        <Button
          icon={this.calendarIcon()}
          id="calendar-btn"
          onClick={this.calenderToggle}
        />
        {this.state.calenderOpen && (
          <Calendar
            sendDateData={this.sendDateData}
            handleCloseModal={this.calenderToggle}
          />
        )}
      </div>
      <span className="question-form__sections">Time</span>
      {this.renderTimeInputs()}
      <span className="question-form__sections">Question</span>
      {this.renderQuestionInputBox()}
      <div>
        <span className="question-form__sections">Question Type</span>
        <Select
          labelText=""
          name="questionType"
          id="selectType"
          value={this.state.questionType}
          onChange={this.handleInputChange}
          wrapperClassName="input-wrapper"
          placeholder="Select Type"
          options={this.state.options}
        />
      </div>
      {this.renderActionButtons()}
    </div>
  );

  render() {
    return (
      <MrmModal
        title="ADD QUESTION"
        buttonText="Add Question"
        closeModal={this.state.closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="add-question-modal"
      >
        <form className="question-form">{this.renderQuestionValues()}</form>
      </MrmModal>
    );
  }
}

export default AddQuestion;
