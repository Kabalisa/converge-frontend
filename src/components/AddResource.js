import React from 'react';
import MrmModal from '../components/commons/Modal';
import '../assets/styles/addresource.scss';

class AddResource extends React.Component {
  state = {
    amenity: '',
    closeModal: false,
  };

  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleModalStateChange = () => {
    if (this.state.closeModal) this.setState({ closeModal: false });
  };

  handleAddAmenity = (event) => {
    event.preventDefault();
    // submission logic goes here
    this.handleCloseModal();
  };

  render() {
    const { amenity, closeModal } = this.state;

    return (
      <MrmModal
        title="ADD AMENITIES"
        buttonText="Add Amenities"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="add-resource-modal"
      >
        <form className="amenity-form" onSubmit={this.handleAddAmenity}>
          <div>
            <label htmlFor="amenity">
              <br />
              Amenity
              <br />
              <br />
              <input
                type="text"
                id="amenity"
                name="amenity"
                value={amenity}
                onChange={this.handleInputChange}
                placeholder="Separate amenities with comma"
              />
            </label>
            <button type="submit">ADD AMENITY</button>
          </div>
        </form>
      </MrmModal>
    );
  }
}

export default AddResource;
