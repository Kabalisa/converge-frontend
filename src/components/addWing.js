/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import toastr from 'toastr';
import MrmModal from '../components/commons/Modal';
import { Input, SelectInput as Select } from './commons';
import '../assets/styles/addoffice.scss';
import ActionButtons from './commons/ActionButtons';
import ADD_WING_MUTATION from '../graphql/mutations/wings';
import GET_FLOORS_QUERY from '../graphql/queries/Floors';
import notification from '../utils/notification';

export class AddWing extends Component {
  static propTypes = {
    addWing: PropTypes.func.isRequired,
  };


  state = {
    name: '',
    floorId: '',
    closeModal: false,
  };

  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  handleAddWing = (event) => {
    event.preventDefault();
    const { name, floorId } = this.state;
    if (!name) {
      notification(toastr, 'error', 'Wing name is required')();
    } else {
      this.props
        .addWing({
          variables: {
            name,
            floorId,
          },
        })
        .then((wing) => {
          notification(
            toastr,
            'success',
            `${wing.data.addWing.wing.name} wing has been added successfully`,
          )();
        })
        .catch((err) => {
          notification(toastr, 'error', err.graphQLErrors[0].message)();
        });
      this.handleCloseModal();
    }
  };

  render() {
    const { name, floorId, closeModal } = this.state;
    const { allFloors } = this.props.allFloors;
    const { handleAddWing, handleInputChange, handleCloseModal } = this;

    return (
      <MrmModal
        title="ADD WING"
        buttonText="Add Wing"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="add-office-modal"
        modalButton="add-button"
      >
        <form className="modal-form" onSubmit={handleAddWing}>
          <Input
            labelName="Wing Name"
            name="name"
            value={name}
            placeholder="Enter wing name"
            id="wingName"
            onChange={handleInputChange}
          />
          <Select
            labelText="Select floor"
            name="floorId"
            id="floorId"
            value={floorId}
            onChange={handleInputChange}
            wrapperClassName="input-wrapper"
            placeholder="Select floor"
            options={allFloors}
          />
          <ActionButtons
            withCancel
            onClickCancel={handleCloseModal}
            actionButtonText="ADD WING"
          />
        </form>
      </MrmModal>
    );
  }
}

export default compose(
  graphql(GET_FLOORS_QUERY, { name: 'allFloors' }),
  graphql(ADD_WING_MUTATION, { name: 'addWing' }),
)(AddWing);
