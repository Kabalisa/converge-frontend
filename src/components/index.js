import NavBar from './NavBar';
import SideBar from './SideBar';
import ContentBlock from './ContentBlock';
import Login from './Login';

export { NavBar, SideBar, ContentBlock, Login };
