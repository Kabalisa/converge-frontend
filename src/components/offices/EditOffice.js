import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import { graphql, compose } from 'react-apollo';
import MrmModal from '../commons/Modal';
import ActionButtons from '../commons/ActionButtons';
import { Input, SelectInput as Select } from '../commons';
import '../../assets/styles/editoffice.scss';
import notification from '../../utils/notification';
import { GET_ALL_OFFICES } from '../../graphql/queries/Offices';
import { EDIT_OFFICE_MUTATION } from '../../graphql/mutations/offices';

/**
 * Edit Office Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
export class EditOffice extends React.Component {
  state = {
    officeName: this.props.officeName,
    officeLocation: this.props.officeLocation,
    officeId: this.props.officeId,
    closeModal: false,
    isLoading: false,
  };

  /**
   * Ensures that the state is updated basing
   * on the changes in the input fields
   *
   * @param {Object} target
   *
   * @returns {void}
   */
  handleInputChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  };

  /**
   * It closes a modal
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * HIt changes the state of the modal
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  /**
   * Handles editing office
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleEditOffice = (event) => {
    event.preventDefault();
    this.toggleLoading();
    const { officeId, officeName } = this.state;
    const { refetch, currentPage } = this.props;
    this.props
      .editOffice({
        variables: {
          officeId,
          name: officeName,
        },
      })
      .then(() => {
        notification(toastr,
          'success', `${officeName} office has been updated successfully`)();
        this.handleCloseModal();
        this.toggles(); refetch({ page: currentPage });
      })
      .catch((err) => {
        this.toggleLoading();
        this.handleCloseModal();
        this.setState({
          officeName: this.props.officeName,
        });
        notification(toastr, 'error', err.graphQLErrors[0].message)();
      });
  };

  /**
   * 1. change isLoading state to it's opposite value
   * i.e true to false or vise verser
   *
   * @returns {void}
   */
  toggleLoading = () => {
    this.setState({
      isLoading: !this.state.isLoading,
    });
  }

  render() {
    const {
      officeName, officeLocation, closeModal, isLoading,
    } = this.state;

    return (
      <MrmModal
        title="EDIT OFFICE"
        buttonText="Edit"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="edit-office-modal"
        modalButtonClassName="edit-button"
      >
        <form className="modal-form">
          <Input
            labelName="Office Name"
            labelClass="label1"
            inputClass="input1"
            name="officeName"
            value={officeName}
            id="officeName"
            onChange={this.handleInputChange}
            required
          />
          <Select
            labelText="Select location"
            name="officeLocation"
            id="officeLocation"
            value={officeLocation}
            onChange={this.handleInputChange}
            options={[]}
            wrapperClassName="edit-office-select"
            placeholder={officeLocation}
            placeholderValue=""
            disabled
            selectInputClassName="edit-office-location default-select"
          />
          <ActionButtons
            withCancel
            onClickCancel={this.handleCloseModal}
            actionButtonText="SAVE CHANGES"
            isLoading={isLoading}
            onClickSubmit={this.handleEditOffice}
          />
        </form>
      </MrmModal>
    );
  }
}

EditOffice.propTypes = {
  editOffice: PropTypes.func.isRequired,
  officeName: PropTypes.string.isRequired,
  officeId: PropTypes.string.isRequired,
  officeLocation: PropTypes.string.isRequired,
  refetch: PropTypes.func,
  currentPage: PropTypes.number,
};

EditOffice.defaultProps = {
  refetch: PropTypes.func,
  currentPage: null,
};

export default compose(
  graphql(GET_ALL_OFFICES, {
    name: 'allOffices',
    options: () => ({
      variables: {
        page: 1,
        perPage: 5,
      },
      options: { refetchQueries: [{ query: GET_ALL_OFFICES }] },
    }),
  }),
  graphql(EDIT_OFFICE_MUTATION, { name: 'editOffice' }),
)(EditOffice);
