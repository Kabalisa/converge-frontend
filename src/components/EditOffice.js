import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import { graphql } from 'react-apollo';
import MrmModal from '../components/commons/Modal';
import ActionButtons from './commons/ActionButtons';
import { Input, SelectInput as Select } from '../components/commons';
import '../assets/styles/editoffice.scss';
import notification from '../utils/notification';
import { GET_ALL_OFFICES } from '../graphql/queries/Offices';
import { EDIT_OFFICE_MUTATION } from '../graphql/mutations/offices';

export class EditOffice extends React.Component {
  state = {
    officeName: this.props.officeName,
    officeLocation: this.props.officeLocation,
    officeId: this.props.officeId,
    closeModal: false,
  };

  handleInputChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  }

  handleCloseModal = () => {
    this.setState({ closeModal: true });
  }

  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  }

  handleEditOffice = (event) => {
    event.preventDefault();
    const { officeId, officeName } = this.state;
    this.props.editOffice({
      variables: {
        officeId,
        name: officeName,
      },
    }).then(() => {
      notification(toastr, 'success', `${officeName} office has been updated successfully`)();
    }).catch((err) => {
      this.setState({
        officeName: this.props.officeName,
      });
      notification(toastr, 'error', err.graphQLErrors[0].message)();
    });
    this.handleCloseModal();
  }

  render() {
    const {
      officeName, officeLocation, closeModal,
    } = this.state;

    return (
      <MrmModal
        title="EDIT OFFICE"
        buttonText="Edit"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="edit-office-modal"
        modalButtonClassName="edit-button"
      >
        <form className="modal-form" onSubmit={this.handleEditOffice}>
          <Input
            labelName="Office Name"
            labelClass="label1"
            inputClass="input1"
            name="officeName"
            value={officeName}
            id="officeName"
            onChange={this.handleInputChange}
            required
          />
          <Select
            labelText="Select location"
            name="officeLocation"
            id="officeLocation"
            value={officeLocation}
            onChange={this.handleInputChange}
            options={[]}
            wrapperClassName="edit-office-select"
            placeholder={officeLocation}
            placeholderValue=""
            disabled
            selectInputClassName="edit-office-location default-select"
          />
          <ActionButtons
            withCancel
            onClickCancel={this.handleCloseModal}
            actionButtonText="SAVE CHANGES"
          />
        </form>
      </MrmModal>
    );
  }
}

EditOffice.propTypes = {
  editOffice: PropTypes.func.isRequired,
  officeName: PropTypes.string.isRequired,
  officeId: PropTypes.string.isRequired,
  officeLocation: PropTypes.string.isRequired,
};

export default graphql(EDIT_OFFICE_MUTATION, { name: 'editOffice', options: { refetchQueries: [{ query: GET_ALL_OFFICES }] } })(EditOffice);
