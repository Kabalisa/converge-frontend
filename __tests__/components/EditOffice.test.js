import React from 'react';
import { shallow } from 'enzyme';
import { EditOffice } from '../../src/components/EditOffice';

describe('EditOffice Component', () => {
  const props = {
    editOffice: jest.fn(),
    officeName: 'Epic',
    officeId: '1',
    officeLocation: 'Lagos',
  };
  let wrapper = shallow(<EditOffice {...props} />);
  const event = {
    preventDefault: jest.fn(),
    target: {
      name: '',
      value: '',
    },
  };

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a form', () => {
    const modalForm = wrapper.find('form');
    expect(modalForm).toHaveLength(1);
  });

  it('handles handleCloseModal()', () => {
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('closeModal')).toEqual(true);
  });

  it('handles handleModalStateChange()', () => {
    wrapper.instance().handleModalStateChange();
    expect(wrapper.state('closeModal')).toEqual(false);
  });

  it('inputs should respond to state changes', () => {
    wrapper.find('#officeName').simulate('change', { target: { name: 'officeName', value: 'Epic building' } });
    expect(wrapper.find('#officeName').props().value).toBe('Epic building');
    expect(wrapper.state('officeName')).toEqual('Epic building');
  });

  it('should handle handleEditOffice() when editOffice is rejected', () => {
    wrapper.setProps({
      editOffice: jest.fn(() => Promise.reject()),
    });
    wrapper.setState({
      officeName: 'Epic Tower',
    });
    wrapper.update();
    wrapper.instance().handleEditOffice(event);
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('closeModal')).toEqual(true);
  });

  it('should handle handleEditOffice() when editOffice is resolved', () => {
    const newProps = {
      editOffice: jest.fn(() => Promise.resolve()),
      officeName: 'Epic',
      officeId: '1',
      officeLocation: 'Lagos',
    };
    wrapper = shallow(<EditOffice {...newProps} />);
    wrapper.instance().handleEditOffice(event);
    expect(newProps.editOffice).toBeCalled();
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('closeModal')).toEqual(true);
  });
});
