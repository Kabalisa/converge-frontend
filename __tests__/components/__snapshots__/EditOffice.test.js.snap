// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditOffice Component renders properly 1`] = `
<MrmModal
  buttonText="Edit"
  className="edit-office-modal"
  closeModal={false}
  handleCloseRequest={[Function]}
  modalButtonClassName="edit-button"
  title="EDIT OFFICE"
>
  <form
    className="modal-form"
    onSubmit={[Function]}
  >
    <Input
      controlsClass=""
      id="officeName"
      inputClass="input1"
      labelClass="label1"
      labelName="Office Name"
      name="officeName"
      onChange={[Function]}
      required={true}
      type="text"
      value="Epic"
    />
    <SelectInput
      disabled={true}
      id="officeLocation"
      labelText="Select location"
      name="officeLocation"
      onChange={[Function]}
      options={Array []}
      placeholder="Lagos"
      placeholderValue=""
      required={false}
      selectInputClassName="edit-office-location default-select"
      value="Lagos"
      wrapperClassName="edit-office-select"
    />
    <ActionButtons
      actionButtonClassName="action-button"
      actionButtonText="SAVE CHANGES"
      cancelButtonClassName="cancel-button"
      cancelButtonText="CANCEL"
      onClickCancel={[Function]}
      withCancel={true}
    />
  </form>
</MrmModal>
`;
